crates/solidity/outputs/cargo/tests/src/bindings_rules.rs

```diff
@@ -0,0 +1,10 @@
+use metaslang_graph_builder::ast::File;
+use slang_solidity::bindings;
+use slang_solidity::cst::KindTypes;
+
+#[test]
+fn test_bindings_rules_parsing() {
+    let graph_builder = File::<KindTypes>::from_str(bindings::get_binding_rules());
```

Comment:
This is a similar situation as the above. This test exists to provide better DX when writing the rules. If the rules fail to parse, everything will be broken and none of the snapshots will be produced. This test is meant to be a prerequisite to that, and to provide more explicit indication of what is wrong with the rules file. As is, this test is not very useful because we only get a failed assertion. I should probably port over a change I made in a couple [other](https://github.com/manastech/slang/blob/feat/query-syntax-changes/crates/solidity/outputs/cargo/tests/src/binding_rules.rs#L12-L21) [branches](https://github.com/manastech/slang/blob/feat/import-resolver/crates/solidity/outputs/cargo/tests/src/binding_rules.rs#L12-L21) to actually show what is the parsing error.

Reply:
Maybe this should be a unit test that lives in the runtime template file? That would allow us to avoid exposing `get_binding_rules()`. But OTOH, we should handle the stubs case and write a valid rules file for `testlang`.

Reply:
Like in #1030, I changed the `get_binding_rules()` to be exposed only when activating the `__private_testing_utils` feature flag.

Reply:
Sounds good.